### PR TITLE
refactor: rename flow task to flow

### PR DESCRIPTION
**Pull request by WenyXu**
I hereby agree to the terms of the [GreptimeDB CLA](https://github.com/GreptimeTeam/.github/blob/main/CLA.md).

## Refer to a related PR or issue link (optional)
#3664
## What's changed and what's your intention?

Wait for #3810 
Remove all `task` suffix.

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.
- [x]  This PR does not require documentation updates.

### CODE REVIEW

File: src/common/meta/src/error.rs [hunk 1]
```diff
@@ -241,9 +241,9 @@ pub enum Error {
         location: Location,
     },
 
-    #[snafu(display("Task already exists: {}", task_name))]
+    #[snafu(display("Task already exists: {}", flow_name))]
     TaskAlreadyExists {
```
Comment (waynexia) [inline src/common/meta/src/error.rs:245]: `FlowAlreadyExists`

File: src/common/meta/src/key/flow/flow_name.rs [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+    flow_id: FlowId,
+}
+
+impl FlowNameValue {
+    /// Returns a [FlowNameValue] with specified [FlowTaskId].
+    pub fn new(flow_id: FlowId) -> Self {
+        Self { flow_id }
+    }
+
+    /// Returns the [FlowTaskId]
```
Comment (waynexia) [inline src/common/meta/src/key/flow/flow_name.rs:122]: here

File: src/common/meta/src/key/flow/flownode_flow.rs [hunk 3]
```diff
@@ -85,8 +85,8 @@ impl FlownodeTaskKey {
     }
 
     /// Returns the [FlowTaskId].
```
Comment (waynexia) [inline src/common/meta/src/key/flow/flownode_flow.rs:87]: here

File: src/common/meta/src/key/flow/flownode_flow.rs [hunk 4]
```diff
@@ -95,91 +95,87 @@ impl FlownodeTaskKey {
     }
 
     /// Returns the [PartitionId].
-    pub fn partition_id(&self) -> FlowTaskPartitionId {
+    pub fn partition_id(&self) -> FlowPartitionId {
         self.0.partition_id
     }
 }
 
 /// The key of mapping [FlownodeId] to [FlowTaskId].
```
Comment (waynexia) [inline src/common/meta/src/key/flow/flownode_flow.rs:103]: here

File: src/common/meta/src/rpc/ddl.rs [hunk 5]
```diff
@@ -729,43 +729,43 @@ impl TryFrom<DropDatabaseTask> for PbDropDatabaseTask {
...
     pub or_replace: bool,
     pub create_if_not_exists: bool,
     pub expire_when: String,
     pub comment: String,
     pub sql: String,
-    pub options: HashMap<String, String>,
+    pub flow_options: HashMap<String, String>,
 }
 
 impl TryFrom<PbCreateFlowTask> for CreateFlowTask {
```
Comment (waynexia) [inline src/common/meta/src/rpc/ddl.rs:747]: here

File: src/common/meta/src/ddl/create_flow.rs [hunk 6]
```diff
@@ -151,7 +151,7 @@ impl Procedure for CreateFlowProcedure {
 
     fn lock_key(&self) -> LockKey {
         let catalog_name = &self.data.task.catalog_name;
-        let task_name = &self.data.task.task_name;
+        let task_name = &self.data.task.flow_name;
```
Comment (evenyag) [inline src/common/meta/src/ddl/create_flow.rs:154]:
```suggestion
        let flow_name = &self.data.task.flow_name;
```

File: src/common/meta/src/ddl/create_flow.rs [hunk 7]
```diff
@@ -206,21 +206,21 @@ impl From<&CreateFlowTaskData> for CreateRequest {
...
     }
 }
 
-impl From<&CreateFlowTaskData> for FlowTaskInfoValue {
+impl From<&CreateFlowTaskData> for FlowInfoValue {
     fn from(value: &CreateFlowTaskData) -> Self {
         let CreateFlowTask {
             catalog_name,
-            task_name,
+            flow_name: task_name,
```
Comment (evenyag) [inline src/common/meta/src/ddl/create_flow.rs:218]: CreateFlowTaskData still uses task_name
```suggestion
            flow_name,
```

File: src/common/meta/src/ddl/create_flow/check.rs [hunk 8]
```diff
@@ -24,20 +24,20 @@ impl CreateFlowProcedure {
     /// - The sink table doesn't exist.
     pub(crate) async fn check_creation(&self) -> Result<()> {
         let catalog_name = &self.data.task.catalog_name;
-        let task_name = &self.data.task.task_name;
+        let task_name = &self.data.task.flow_name;
```
Comment (evenyag) [inline src/common/meta/src/ddl/create_flow/check.rs:27]:
```suggestion
        let flow_name = &self.data.task.flow_name;
```

File: src/common/meta/src/ddl/flow_meta.rs [hunk 9]
```diff
@@ -17,43 +17,43 @@ use std::sync::Arc;
...
 use crate::sequence::SequenceRef;
 
-/// The reference of [FlowTaskMetadataAllocator].
-pub type FlowTaskMetadataAllocatorRef = Arc<FlowTaskMetadataAllocator>;
+/// The reference of [FlowMetadataAllocator].
+pub type FlowMetadataAllocatorRef = Arc<FlowMetadataAllocator>;
 
-/// [FlowTaskMetadataAllocator] provides the ability of:
+/// [FlowMetadataAllocator] provides the ability of:
 /// - [FlowTaskId] Allocation.
```
Comment (evenyag) [inline src/common/meta/src/ddl/flow_meta.rs]:
```suggestion
/// - [FlowId] Allocation.
```

File: src/common/meta/src/ddl/flow_meta.rs [hunk 10]
```diff
@@ -17,43 +17,43 @@ use std::sync::Arc;
...
 /// - [FlownodeId] Selection.
```
Comment (evenyag) [inline src/common/meta/src/ddl/flow_meta.rs:29]: Should it be FlowNode?

File: src/common/meta/src/key.rs [hunk 11]
```diff
@@ -36,16 +36,16 @@
...
 //!     - Used in the table name to table id lookup.
 //!
-//! 6. Flow task info key: `__flow_task/{catalog}/info/{flow_task_id}`
-//!     - Stores metadata of the flow task.
+//! 6. Flow info key: `__flow/{catalog}/info/{flow_id}`
+//!     - Stores metadata of the flow.
 //!
-//! 7. Flow task name key: `__flow_task/{catalog}/name/{task_name}`
-//!     - Mapping {catalog}/{task_name} to {flow_task_id}
+//! 7. Flow name key: `__flow/{catalog}/name/{task_name}`
```
Comment (evenyag) [inline src/common/meta/src/key.rs:42]: Should it be task name?

File: src/common/meta/src/key/flow.rs [hunk 12]
```diff
@@ -79,140 +79,134 @@ impl<T: MetaKey<T>> MetaKey<FlowTaskScoped<T>> for FlowTaskScoped<T> {
...
-        &self.flow_task_name_manager
+    /// Returns the [FlowNameManager].
+    pub fn flow_name_manager(&self) -> &FlowNameManager {
+        &self.flow_name_manager
     }
 
-    /// Returns the [FlowTaskInfoManager].
-    pub fn flow_task_info_manager(&self) -> &FlowTaskInfoManager {
-        &self.flow_task_info_manager
+    /// Returns the [FlowTaskManager].
```
Comment (evenyag) [inline src/common/meta/src/key/flow.rs:117]:
```suggestion
    /// Returns the [FlowInfoManager].
```

File: src/common/meta/src/key/flow/table_flow.rs [hunk 13]
```diff
@@ -111,99 +111,99 @@ impl TableTaskKey {
...
-/// The manager of [TableTaskKey].
-pub struct TableTaskManager {
+/// The manager of [TableFlowKey].
+pub struct TableFlowManager {
     kv_backend: KvBackendRef,
 }
 
-impl TableTaskManager {
+impl TableFlowManager {
     /// Returns a new [TableTaskManager].
```
Comment (evenyag) [inline src/common/meta/src/key/flow/table_flow.rs]:
```suggestion
    /// Returns a new [TableFlowManager].
```

File: src/common/meta/src/key/flow/table_flow.rs [hunk 14]
```diff
@@ -111,99 +111,99 @@ impl TableTaskKey {
...
             partition_id,
         ))
     }
 }
 
-/// Decodes `KeyValue` to [TableTaskKey].
-pub fn table_task_decoder(kv: KeyValue) -> Result<TableTaskKey> {
-    TableTaskKey::from_bytes(&kv.key)
+/// Decodes `KeyValue` to [TableFlowKey].
+pub fn table_task_decoder(kv: KeyValue) -> Result<TableFlowKey> {
```
Comment (evenyag) [inline src/common/meta/src/key/flow/table_flow.rs]:
```suggestion
pub fn table_flow_decoder(kv: KeyValue) -> Result<TableFlowKey> {
```

File: src/common/meta/src/key/flow/table_flow.rs [hunk 15]
```diff
@@ -218,10 +218,10 @@ impl TableTaskManager {
     /// Builds a create table task transaction.
     ///
     /// Puts `__table_task/{table_id}/{node_id}/{partition_id}` keys.
-    pub fn build_create_txn<I: IntoIterator<Item = (FlowTaskPartitionId, FlownodeId)>>(
+    pub fn build_create_txn<I: IntoIterator<Item = (FlowPartitionId, FlownodeId)>>(
```
Comment (evenyag) [inline src/common/meta/src/key/flow/table_flow.rs]: Should it be table_flow?

File: src/common/meta/src/ddl/flow_meta.rs (hunk 10 above)
Comment (WenyXu) [reply src/common/meta/src/ddl/flow_meta.rs:29]: Follow the style of `DatanodeId`

**Review by evenyag — APPROVED** [review]

**Pull request merged**